src/utilities/updateBrowserUrl.browser.spec.ts

```diff
@@ -0,0 +1,42 @@
+
+import { expect, test, vi } from 'vitest'
+import { updateBrowserUrl } from '@/utilities/updateBrowserUrl'
+
+test('updates the window location', () => {
+  vi.spyOn(window.location, 'assign').mockImplementation(() => {})
+  const url = 'http://example.com2/foo'
+
+  updateBrowserUrl(url)
+
+  expect(window.location.assign).toHaveBeenCalledWith(url)
+})
+
+test('updates the window location using replace when options.replace is true', () => {
+  vi.spyOn(window.location, 'replace').mockImplementation(() => {})
+
+  const url = 'http://example.com/foo'
+
+  updateBrowserUrl(url, { replace: true })
+
+  expect(window.location.replace).toHaveBeenCalledWith(url)
+})
+
+test('updates the history', () => {
+  vi.spyOn(history, 'pushState').mockImplementation(() => {})
+
+  const url = '/foo'
+
+  updateBrowserUrl(url)
+
+  expect(history.pushState).toHaveBeenCalledWith({}, '', url)
+})
```

Comment:
I feel like there is some overlap here with the service I'm working on. It might make sense in the future to update this service/test to call our local history service and have tests over there that ensure that it properly sends updated state over to browser history as well